src/keystore/persistence/LocalStoragePonyfill.ts

```diff
@@ -0,0 +1,62 @@
+// Fully in-memory polyfill for the browser storage API.
+// Borrowed from https://github.com/MitchellCash/node-storage-polyfill but implemented as a ponyfill instead of a polyfill
+
+export default class LocalStoragePonyfill implements Storage {
+  store: Map<string, string>
+  constructor() {
+    this.store = new Map()
+  }
+
+  get length(): number {
+    return this.store.size
```

Comment:
I have the worst suggestion ever: Would it be worth tracking a `total_bytes` parameter and having a `byte_size` method? Might be useful for metrics and stuff but probably not worth the bookkeeping right now.

Reply:
So, this ponyfill is meant to be a 1:1 match of the browser `Storage` API.

If we wanted to track `total_bytes` I'd be tempted to do that bookkeeping on top of the real `LocalStorage`, which does have some hard limits (5mb total).

Reply:
Oh that makes a lot more sense